src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs

```diff
@@ -317,18 +317,20 @@ internal void Connect(ServerInfo serverInfo, SqlInternalConnectionTds connHandle
                 Debug.Assert(false, "SNI returned status != success, but no error thrown?");
             }
 
-#if !MANAGED_SNI
+
             if (integratedSecurity)
             {
+#if !MANAGED_SNI
                 LoadSSPILibrary();
+#endif
                 // now allocate proper length of buffer
                 _sniSpnBuffer = new byte[SNINativeMethodWrapper.SniMaxComposedSpnLength];
```

Comment:
You can change the method to pass a boolean. The problem with this approach is that you are allocating memory and then allocating more later when the SPN is created.